### PR TITLE
[flags] Cleanup enableCache

### PR DESCRIPTION
**Pull request by rickhanlonii**
This is landed everywhere

### CODE REVIEW

**Comment by sebmarkbage** [comment]
I we used to have an `enableCacheElement` flag and I'm not sure when that was deleted. However, that flag was not properly flagging everything so now we're left which a bunch of dead code.

https://github.com/facebook/react/blob/main/packages/react-reconciler/src/ReactFiberCommitWork.js#L2906-L2920

We probably need to also remove the `CacheComponent` work.

**Comment by rickhanlonii** [comment]
@sebmarkbage yeah I was going to ask about that because a ton of this looks unnecessary, do you thing I should followup to remove that dead code so I can revert it cleanly if needed, or just delete it in this PR?

File: packages/react-reconciler/src/ReactFiberBeginWork.js [hunk 1]
```diff
@@ -3797,10 +3778,8 @@ function attemptEarlyBailoutIfNoScheduledUpdate(
       return updateOffscreenComponent(current, workInProgress, renderLanes);
     }
     case CacheComponent: {
-      if (enableCache) {
-        const cache: Cache = current.memoizedState.cache;
-        pushCacheProvider(workInProgress, cache);
-      }
+      const cache: Cache = current.memoizedState.cache;
```
Comment (sebmarkbage) [inline packages/react-reconciler/src/ReactFiberBeginWork.js:3781]: Need to fix these lints. Interesting that it wasn't failing for shadowing.

**Review by sebmarkbage — APPROVED** [review]

File: packages/react-reconciler/src/ReactFiberBeginWork.js (hunk 1 above)
Comment (rickhanlonii) [reply packages/react-reconciler/src/ReactFiberBeginWork.js:3782]: TIL case statements without braces do not scope let/const to that case, fixed by adding braces to the previous case.

**Pull request merged**